src/python_testing/TC_CLDIM_5_1.py

```diff
@@ -0,0 +1,242 @@
+        if attributes.LimitRange.attribute_id in attribute_list:
+            limit_range = await self.read_cldim_attribute_expect_success(endpoint=endpoint, attribute=attributes.LimitRange)
+            min_position = limit_range.Min
+            max_position = limit_range.Max
+
+        # STEP 3a: If manual latching is required, skip steps 3b and 3c
+        self.step("3a")
+        if self.check_pics("CLDIM.S.M.ManualLatching"):
+            logging.info("Manual latching is required. Skipping steps 3b and 3c.")
+            self.skip_step("3b")
+            self.skip_step("3c")
+
+        # STEP 3b: Send SetTarget command with Latch=True
+        self.step("3b")
+        try:
+            await self.send_single_cmd(
+                cmd=Clusters.Objects.ClosureDimension.Commands.SetTarget(Latch=True),
+                endpoint=endpoint
+            )
+
+            asserts.fail("Expected InvalidAction error, but command succeeded")
+
+        except InteractionModelError as e:
+            asserts.assert_equal(e.status, Status.InvalidAction, "Unexpected status returned")
+
+        # STEP 3c: Manually latch the device
+        self.step("3c")
+        test_step = "Manually latch the device"
+        self.wait_for_user_input(prompt_msg=f"{test_step}, and press Enter when ready.")
+
+        # STEP 4a: If manual latching is not required, skip steps 4b to 4d
+        self.step("4a")
+        if not self.check_pics("CLDIM.S.M.ManualLatching"):
+            self.skip_step("4b")
+            self.skip_step("4c")
+            self.skip_step("4d")
+
+        # STEP 4b: Send SetTarget command with Latch=True
+        self.step("4b")
+        try:
+            await self.send_single_cmd(
+                cmd=Clusters.Objects.ClosureDimension.Commands.SetTarget(Latch=True),
+                endpoint=endpoint
+            )
+        except InteractionModelError as e:
+            asserts.assert_equal(e.status, Status.Success, "Unexpected error returned")
+
+        # STEP 4c: Verify Target attribute is updated
+        self.step("4c")
+        if attributes.Target.attribute_id in attribute_list:
+            target = await self.read_cldim_attribute_expect_success(endpoint=endpoint, attribute=attributes.Target)
+
+            if is_positioning_supported:
+                asserts.assert_greater_equal(target.Position, min_position, "Target Position is outside expected range")
+                asserts.assert_less_equal(target.Position, max_position, "Target Position is outside expected range")
+            asserts.assert_equal(target.Latch, True, "Target Latch is not True")
+
+            if is_speed_supported:
+                asserts.assert_greater_equal(target.Speed, 0, "Target Speed is outside allowed range")
+                asserts.assert_less_equal(target.Speed, 3, "Target Speed is outside allowed range")
+
+        # STEP 4d: Wait for PIXIT.CLDIM.LatchingDuration seconds
+        self.step("4d")
+        time.sleep(latching_duration)
```

Comment:
The test should wait for the reporting of latch state change through subscription on a time-out instead of sleeping, Then after the report, a manual step should ask whether the closure is actually latched. This would both allow testing subscriptions AND remove manual setup in PIXIT.